server/app/services/geo/esri/EsriServiceAreaValidationConfig.java

```diff
@@ -0,0 +1,146 @@
+        || this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_ATTRIBUTES.isEmpty()) {
+      return false;
+    } else {
+      // check for same size
+      if (this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_LABELS.get().size()
+              == this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_IDS.get().size()
+          && this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_IDS.get().size()
+              == this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_URLS.get().size()
+          && this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_URLS.get().size()
+              == this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_ATTRIBUTES.get().size()) {
+        return true;
+      } else {
+        return false;
+      }
+    }
+  }
+
+  private Optional<ImmutableList<String>> getConfigListSetting(Optional<ConfigList> setting) {
+    if (setting.isEmpty()) {
+      logger.error(CONFIG_ERROR_STRING);
+      return Optional.empty();
+    }
+
+    ImmutableList.Builder<String> listBuilder = ImmutableList.builder();
+    setting.get().stream()
+        .forEach(configValue -> listBuilder.add((String) configValue.unwrapped()));
+    return Optional.of(listBuilder.build());
+  }
+
+  /**
+   * Transforms the config settings for Esri address service area validation into an optional
+   * immutable map with a service area ID as the key and {@link EsriServiceAreaValidationOption} as
+   * the value.
+   */
+  public Optional<ImmutableMap<String, EsriServiceAreaValidationOption>> getImmutableMap() {
+    if (this.esriServiceAreaValidationMap != null) {
+      return this.esriServiceAreaValidationMap;
+    }
+
+    if (!hasAllElements()) {
+      logger.error(CONFIG_ERROR_STRING);
+      return Optional.empty();
+    }
+
+    ImmutableMap.Builder<String, EsriServiceAreaValidationOption> mapBuilder =
+        ImmutableMap.builder();
+
+    ImmutableList<String> labels =
+        getConfigListSetting(this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_LABELS).get();
+    ImmutableList<String> values =
+        getConfigListSetting(this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_IDS).get();
+    ImmutableList<String> urls =
+        getConfigListSetting(this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_URLS).get();
+    ImmutableList<String> attributes =
+        getConfigListSetting(this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_ATTRIBUTES).get();
+
+    for (int i = 0; i < values.size(); i++) {
+      EsriServiceAreaValidationOption option =
+          EsriServiceAreaValidationOption.builder()
+              .setLabel(labels.get(i))
+              .setId(values.get(i))
+              .setUrl(urls.get(i))
+              .setAttribute(attributes.get(i))
+              .build();
+
+      mapBuilder.put(values.get(i), option);
+    }
+    return Optional.of(mapBuilder.build());
+  }
+
+  /**
+   * Returns an {@link EsriServiceAreaValidationOption} from the config list for Esri address
+   * service area settings given a service area id.
+   */
+  public Optional<EsriServiceAreaValidationOption> getOptionByServiceAreaId(String serviceAreaId) {
```

Comment:
@bion , this isn't getting used by EsriClient anymore. Do you think it'll be useful in attaching service area validation to predicates? If not, I can remove it.

Reply:
I say keep it. When the admin specifies a service area predicate we need to look up the service area by ID to ensure their form submission is valid before persisting.